common/lib/plugins/efm2/host_monitoring2_connection_plugin.ts

```diff
@@ -0,0 +1,175 @@
+import { HostInfo } from "../../host_info";
+import { OldConnectionSuggestionAction } from "../../old_connection_suggestion_action";
+import { RdsUtils } from "../../utils/rds_utils";
+import { AbstractConnectionPlugin } from "../../abstract_connection_plugin";
+import { RdsUrlType } from "../../utils/rds_url_type";
+import { WrapperProperties } from "../../wrapper_property";
+import { MonitorConnectionContext } from "./monitor_connection_context";
+import { logger, uniqueId } from "../../../logutils";
+import { Messages } from "../../utils/messages";
+import { MonitorService, MonitorServiceImpl } from "./monitor_service";
+import { AwsWrapperError } from "../../utils/errors";
+import { HostListProvider } from "../../host_list_provider/host_list_provider";
+import { CanReleaseResources } from "../../can_release_resources";
+import { SubscribedMethodHelper } from "../../utils/subscribed_method_helper";
+import { ClientWrapper } from "../../client_wrapper";
+
+export class HostMonitoring2ConnectionPlugin extends AbstractConnectionPlugin implements CanReleaseResources {
+  id: string = uniqueId("_efm2Plugin");
+  private readonly properties: Map<string, any>;
+  private pluginService: PluginService;
+  private rdsUtils: RdsUtils;
+  private monitoringHostInfo: HostInfo | null = null;
+  private monitorService: MonitorService;
+
+  constructor(pluginService: PluginService, properties: Map<string, any>, rdsUtils: RdsUtils = new RdsUtils(), monitorService?: MonitorServiceImpl) {
+    super();
+    this.pluginService = pluginService;
+    this.properties = properties;
+    this.rdsUtils = rdsUtils;
+    this.monitorService = monitorService ?? new MonitorServiceImpl(pluginService);
+  }
+
+  getSubscribedMethods(): Set<string> {
+    return new Set<string>(["*"]);
+  }
+
+  connect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    connectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    return this.connectInternal(hostInfo, connectFunc);
+  }
+
+  forceConnect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    forceConnectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    return this.connectInternal(hostInfo, forceConnectFunc);
+  }
+
+  private async connectInternal(hostInfo: HostInfo, connectFunc: () => Promise<ClientWrapper>): Promise<ClientWrapper> {
+    const targetClient = await connectFunc();
+    if (targetClient != null) {
+      const type: RdsUrlType = this.rdsUtils.identifyRdsType(hostInfo.host);
+      if (type.isRdsCluster) {
+        hostInfo.resetAliases();
+        await this.pluginService.fillAliases(targetClient, hostInfo);
+      }
+    }
+    return targetClient;
+  }
+
+  async execute<T>(methodName: string, methodFunc: () => Promise<T>, methodArgs: any): Promise<T> {
+    const isEnabled: boolean = WrapperProperties.FAILURE_DETECTION_ENABLED.get(this.properties) as boolean;
```

Comment:
I don't think the casting is required
```suggestion
    const isEnabled: boolean = WrapperProperties.FAILURE_DETECTION_ENABLED.get(this.properties);
```